pkg/vif/routing/routing_darwin.go

```diff
@@ -57,16 +57,16 @@ func GetRoutingTable(ctx context.Context) ([]*Route, error) {
 			if !ok {
 				continue
 			}
-			gw, ok := gw.(*route.Inet4Addr)
-			if !ok {
-				continue
+			var gwIP net.IP
+			if gwAddr, ok := gw.(*route.Inet4Addr); ok {
+				gwIP = gwAddr.IP[:]
 			}
 			routes = append(routes, &Route{
 				Interface: iface,
-				Gateway:   net.IP(gw.IP[:]),
+				Gateway:   gwIP,
```

Comment:
So this can now be nil ?

Reply:
It means that the gateway is of type link (to another interface), so there's no gateway IP. E.g. a VPN uses an interface and then also manages other subnets.